src/util/urlHelper.ts

```diff
@@ -0,0 +1,64 @@
+/**
+ * Copyright (c) 2017-present, NDLA.
+ *
+ * This source code is licensed under the GPLv3 license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ */
+
+import { matchPath } from 'react-router-dom';
+import Url from 'url-parse';
+import { match as RouterMatchType } from 'react-router';
+import { isValidLocale } from '../i18n';
+import {
+  PLAIN_ARTICLE_PAGE_PATH,
+  RESOURCE_PAGE_PATH,
+  TOPIC_PATH,
+  PLAIN_ARTICLE_IFRAME_PATH,
+  RESOURCE_ARTICLE_IFRAME_PATH,
+  TOPIC_ARTICLE_IFRAME_PATH,
+} from '../constants';
+
+function matchUrl<Params>(
+  pathname: string,
+  lang: boolean = false,
+): RouterMatchType<Params> | null {
+  const possiblePaths = [
+    lang ? `/:lang${RESOURCE_PAGE_PATH}` : RESOURCE_PAGE_PATH,
+    lang ? `/:lang${TOPIC_PATH}` : TOPIC_PATH,
+    lang ? `/:lang${PLAIN_ARTICLE_PAGE_PATH}` : PLAIN_ARTICLE_PAGE_PATH,
+    lang ? `/:lang${PLAIN_ARTICLE_IFRAME_PATH}` : PLAIN_ARTICLE_IFRAME_PATH,
+    lang
+      ? `/:lang${RESOURCE_ARTICLE_IFRAME_PATH}`
+      : RESOURCE_ARTICLE_IFRAME_PATH,
+    lang ? `/:lang${TOPIC_ARTICLE_IFRAME_PATH}` : TOPIC_ARTICLE_IFRAME_PATH,
+  ];
+
+  for (let i = 0; i < possiblePaths.length; i++) {
+    const attempt = possiblePaths[i] ?? '';
+    const m = matchPath<Params>(pathname, attempt);
+    if (m) {
+      return m;
+    }
+  }
```

Comment:
```suggestion
  possiblePaths?.find(path => {
    const m = matchPath<Params>(pathname, path);
    if (m) {
      return m;
    }    
  });
```

Reply:
nja, da må vi i så fall kalle på `matchPath` 2 ganger for parameteret som treffer, siden vi er ute etter `match` objektet som kommer fra `matchPath` og ikke bare hvilken path som matcher.

Reply:
Åja. Det er sant.

Reply:
`find` fungerer vel på den måten at den returnerer første elementet i arrayen hvor callbacken returnerer noe truthy:

```javascript
const arr = [null, 'yes', 'no'];
const res = arr.find(e => {
  if(e)
    return 'noeheltannet';
});
console.log(res);
> 'yes'
```

Så vi vil bare få første pathen som returnerte noe truthy og må kalle `matchPath` en gang til i så fall.


Vi kan gjøre:
```typescript
possiblePaths
  .map(p => matchPath<Params>(pathname, p))
  .find(e => !!e);
```

Men siden `map` er greedy så vil vi kjøre `matchPath` for alle verdiene når vi egentlig bare vil ha den første som matcher.